python/graphstorm/dataloading/dataloading.py

```diff
@@ -887,3 +894,165 @@ def __iter__(self):
 
     def __next__(self):
         return self.dataloader.__next__(), self.unlabeled_dataloader.__next__()
+
+####################### Distillation #############################
```

Comment:
all data loaders and dataset classes should have unit tests. the fact that these classes have no tests is concerning. the last time we didn't enforce the policy of having unit tests for the code leads to serious technical debt in DGL's distributed graph partitioning pipeline.

Reply:
added.